### PR TITLE
Validations specs

### PR DESCRIPTION
**Pull request by initseis**
## [IMPORTANT] You have to have data in your database to render successfully
- Use my seed file with the command `rails db:seed` to create a user and then start adding posts, comments, and likes
- Style is not a requirement yet
- Current user is the first user as the requirements asked
- Requirements ask PostsCounter, CommentsCounter, LikesCounter have to be greater than zero but when you create a user, It doesn't have posts. I think that's a mistake and I used greater than or equal to zero as in the issues suggest.

## Description
In this branch I:

- Added the following validations:
  - For the User model:
    - Name must not be blank.
    - PostsCounter must be an integer greater than zero.
    
       imagen

  - For the Post model:
    - Title must not be blank.
    - Title must not exceed 250 characters.
    - CommentsCounter must be an integer greater than zero.
    - LikesCounter must be an integer greater than zero.
    
       imagen

- Added unit specs for all of my models' methods and validations. (run `rspec spec/models/*.rb`)
   
   imagen

- Added flash messages in the create actions in all your controllers.
   
   imagen

- Made sure that the N+1 problem is solved when fetching all posts and their comments for a user.
  - I solved the N+1 problem for the comments
   imagen

 ## No less important

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -1,5 +1,24 @@
...
+  end
+
+  it 'name should be present' do
+    @user.name = nil
+    expect(@user).to_not be_valid
+  end
+
+  it 'posts_counter should be greater than zero' do
+    @user.posts_counter = 0
+    expect(@user).to_not be_valid
```
Comment (Davidosky007) [inline spec/models/user_spec.rb]: * Kindly fix this test as it is failing when i run `rspec spec/models/*.rb`
image

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
## Hi @sevinchek 👋, 

Good job so far! 💪 
* Well documented PR ✔️ 
* No linter errors ✔️ 
* Descriptive commit messages ✔️ 
* Nice job adding all the necessary validations and flash messages  ✔️ 
There are some issues that you still need to work on to go to the next project but you are almost there!


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/user_spec.rb (hunk 1 above)
Comment (initseis) [reply spec/models/user_spec.rb]: Thanks for the review, @Davidosky007. I already made the changes.

**Review by arslanbisharat — APPROVED** [review]
# Approved 👏 

Hi there, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

To Highlight: 
- The Pull Request has a proper title and description. ✔️ 
- You have added a descriptive Readme file. Good Job. ✔️ 
- There is no linter error is present. ✔️ 
- Project has met all the technical aspects you needed for this project. Keep rocking! ✔️ 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of code reviews per project which you can see from your dashboard. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)_

**Comment by initseis** [comment]
Thanks, @arslanbisharat @Davidosky007.

**Pull request merged**